addons/mail/static/tests/chat_window_test.js

```diff
@@ -0,0 +1,46 @@
+odoo.define('mail.chat_window_test', function (require) {
+"use strict";
+
+var testUtils = require('web.test_utils');
+var Widget = require('web.Widget');
+var ChatWindow = require('mail.ExtendedChatWindow');
+
+QUnit.module('mail', {}, function () {
+
+    QUnit.test('chat_window: close using ESCAPE key', function (assert) {
+    assert.expect(2);
+
+    function createParent(params) {
+        var widget = new Widget();
+
+        testUtils.addMockEnvironment(widget, params);
+        return widget;
+    }
+    var messages = [{
+        attachment_ids: [],
+        author_id: ["1", "John Doe"],
+        body: "A message",
+        date: moment("2016-12-20 09:35:40"),
+        displayed_author: "John Doe",
+        id: 1,
+        is_note: false,
+        is_starred: false,
+        model: 'partner',
+        res_id: 2
+    }];
+    var parent = createParent({
+        data: {},
+    });
+
+    var chat_window = new ChatWindow(parent, 1, "user", false, messages, {});
+    chat_window.appendTo($('#qunit-fixture'));
+
+    chat_window.thread.$el.trigger("click");
+    assert.equal(chat_window.$input.is(":focus"), true, "Focused");
+    var upKeyEvent = jQuery.Event( "keyup", {which: 27});
+    chat_window.$('.o_composer_input').trigger(upKeyEvent);
+    assert.strictEqual(chat_window.folded, false, "Closed chat Window");
+});
```

Comment:
you forgot to destroy the chatwindow at the end of the test.  This is a memory leak, but is easily fixed: add chatWindow.destroy(); just before the end of the test